### PR TITLE
LibJS: Temporal+++

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp [hunk 1]
```diff
@@ -765,52 +765,31 @@ bool ISO8601Parser::parse_time_zone_utc_offset()
...
 {
-    // TimeZoneNumericUTCOffsetNotAmbiguous :
-    //     + TimeZoneUTCOffsetHour
-    //     U+2212 TimeZoneUTCOffsetHour
-    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour : TimeZoneUTCOffsetMinute
-    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour TimeZoneUTCOffsetMinute
-    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour : TimeZoneUTCOffsetMinute : TimeZoneUTCOffsetSecond TimeZoneUTCOffsetFraction[opt]
-    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour TimeZoneUTCOffsetMinute TimeZoneUTCOffsetSecond TimeZoneUTCOffsetFraction[opt]
+    // FIXME: They likely meant TimeZoneNumericUTCOffset instead of TimeZoneUTCOffset.
+    //        See: https://github.com/tc39/proposal-temporal/issues/2342
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:772]: Fix pending - https://github.com/tc39/proposal-temporal/pull/2343

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp [hunk 2]
```diff
@@ -768,37 +768,16 @@ bool ISO8601Parser::parse_time_zone_utc_offset()
...
-            if (!parse_time_zone_utc_offset_minute())
-                return false;
-            if (parse_time_zone_utc_offset_second())
-                (void)parse_time_zone_utc_offset_fraction();
-        }
-    }
+    if (!parse_time_zone_numeric_utc_offset())
+        return false;
+    auto parsed = *m_state.parse_result.time_zone_numeric_utc_offset;
+    if (parsed.length() == 3 && parsed[0] == '-')
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:779]: U+2212 is included, check the old subproductions :)

[the remark this answers is not in the document]
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:779]: Huh, that's weird, the name is `TimeZoneNumericUTCOffsetNotAmbiguousWithDayOfMonth`, but I'm not entirely sure how that is ambiguous with days of the month.

**Review by IdanHo — APPROVED** [review]
LGTM, up to you if you want to wait until your Temporal PR is merged

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:779]: The original commit where it was introduced might be helpful: https://github.com/tc39/proposal-temporal/commit/514ede3bdfe8f45cc4da0418bee5fb0738fe6d1c

**Pull request merged**